src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let startIndex = 0;
+
+    if (arguments.length < 2) {
+      prev = this[0];
+      startIndex = 1;
+    } else if (typeof initialValue === 'undefined') {
+      prev = `undefined`;
+    }
```

Comment:
extra check, it will work without it
 if 'undefined' it already is in prev (at line 8)